### PR TITLE
3.30.0

### PR DESCRIPTION
**Pull request by trentm**
Release version 3.30.0.  It has been a month. This one has a couple significant changes:

- The capturing of span stacktraces is now *disabled* by default for perf reasons. The `spanFramesMinDuration` config var obsoletes `captureSpanStackTraces` and `spanStackTraceMinDuration` (the old ones still work). https://github.com/elastic/apm-agent-nodejs/pull/2565 for details.
- Instrumentating Lambda functions:
    - One must use [v0.0.4](https://github.com/elastic/apm-aws-lambda/releases/tag/v0.0.4) of the Elastic APM Lambda extension or later, because new functionality in this version of the APM agent tickles a crash bug in older versions of the extension.
    - Instrumenting your Lambda functions is now easier: 1. add the Node.js APM agent layer to your lambda functions, 2. set the `NODE_OPTIONS=-r elastic-apm-node/start` environment variable, and then the manual `apm.lambda(...)` wrapping of your handler function is no longer necessary!

### CODE REVIEW

**Review by astorm — APPROVED** [review]
👍 /approved pending the usual "passes CI" shenanigans

**Comment by trentm** [comment]
v3.30.0 tag pushed
elastic-apm-node@3.30.0 published

**Comment by trentm** [comment]
https://apm-ci.elastic.co/blue/organizations/jenkins/apm-agent-nodejs%2Fapm-agent-nodejs-mbp/detail/main/75/pipeline is the build to watch to see if the "Dist" and "Publish" steps work for the Lambda layers.

**Comment by trentm** [comment]
Or I am wrong. Build https://apm-ci.elastic.co/blue/organizations/jenkins/apm-agent-nodejs%2Fapm-agent-nodejs-mbp/detail/main/75/pipeline/ did *not* run the Dist and Publish steps. I am not sure why.

The change that was meant to make this work was: https://github.com/elastic/apm-agent-nodejs/pull/2585

Perhaps it is a timing issue?
- The PR for the release gets merged to main -> triggers Jenkins build of that commit
- A short time after (per the [release steps](https://github.com/elastic/apm-agent-nodejs/blob/main/CONTRIBUTING.md#current-major)), the tag is released and pushed.

So perhaps that triggered Jenkins build didn't see a tag yet.  Timing in this case was:

- Jenkins build of main begins at 10-Mar-2022 6:26:57 PM (https://apm-ci.elastic.co/job/apm-agent-nodejs/job/apm-agent-nodejs-mbp/job/main/75/)
- The tag and push of the tag at "10:28:59" localtime, which is 18:28:59 UTC, which is 2m02s *after* the start of the above build.

```
[10:28:46 trentm@pink:~/pristine/apm-agent-nodejs (git:main)]
% git tag v3.30.0
[10:28:55 trentm@pink:~/pristine/apm-agent-nodejs (git:main)]
% git push origin !$
git push origin v3.30.0
Total 0 (delta 0), reused 0 (delta 0), pack-reused 0
To github.com:elastic/apm-agent-nodejs.git
 * [new tag]           v3.30.0 -> v3.30.0
[10:28:59 trentm@pink:~/pristine/apm-agent-nodejs (git:main)]
```

I have started a new main build to see if that catches the tag: https://apm-ci.elastic.co/job/apm-agent-nodejs/job/apm-agent-nodejs-mbp/job/main/76/

If so, then we'll have to adjust the release process to ensure we get a "Dist/Release" of the release commit. If there is another commit to main immediately after the merge to main, then we might totally miss it.

**Comment by trentm** [comment]
Nope, build 76 (https://apm-ci.elastic.co/blue/organizations/jenkins/apm-agent-nodejs%2Fapm-agent-nodejs-mbp/detail/main/76/pipeline) skipped the Dist and Publish steps as well. I don't know what is going on.

@v1v @astorm Are you able to help determine why we did not get a Jenkins build of the APM Agent Lambda Layer after the v3.30.0 tag that was added for this release? My understanding from https://github.com/elastic/apm-agent-nodejs/pull/2585 is that this should have happened automatically after I pushed the tag. Or is there a separate step I need to take?

**Comment by trentm** [comment]
Lambda layer builds here: https://github.com/elastic/apm-agent-nodejs/releases/tag/v3.30.0
(after the build fix in #2609)

**Pull request merged**
